packages/starlight/utils/route-data.ts

```diff
@@ -34,8 +34,6 @@ export interface StarlightRouteData extends Route {
 	lastUpdated: Date | undefined;
 	/** URL object for the address where this page can be edited if enabled. */
 	editUrl: URL | undefined;
-	/** Record of UI strings localized for the current page. */
-	labels: ReturnType<ReturnType<typeof useTranslations>['all']>;
 }
```

Comment:
Making a quick note before I forget of an idea to maybe make migration a bit easier for people.

1. Keep the type here and mark it as deprecated.

2. Create a proxy that throws a helpful error when accessing `labels`.

**N.B. `AstroError` will need adding to imports**

```suggestion
	/** @deprecated Use `Astro.locals.t()` instead **/
	labels: Record<string, never>;
}

// TODO: remove in a future release once people have updated. No later than v1.
const labels: Record<string, never> = new Proxy(
	{},
	{
		get(_, key) {
			const label = String(key);
			throw new AstroError(
				`The \`labels\` prop in component overrides has been removed.`,
				`Replace \`labels["${label}"]\` with \`Astro.locals.t("${label}")\` instead.\n` +
					'For more information see https://starlight.astro.build/guides/i18n/#accessing-ui-strings'
			);
		},
	}
);
```